services/website-api/src/controller/authentication/AuthenticationController.ts

```diff
@@ -0,0 +1,59 @@
+import { AuthenticationService } from '@donategifts/authentication';
+import { Controller, Post, Route, Tags, Response, Body } from '@tsoa/runtime';
+import { IUser, ObjectId } from '@donategifts/common';
+// import * as express from 'express';
+import { IAPIUser } from '../user/types/IAPIUser';
+import { IAPISignupResponse } from './types/IAPISignupResponse';
+
+@Route('/auth')
+@Tags('authentication')
+export class AuthenticationController extends Controller {
+	constructor(private authenticationService: typeof AuthenticationService = AuthenticationService) {
+		super();
+	}
+
+	@Response('400', 'Bad request')
+	@Post('/signup')
+	public async signup(
+		@Body() body: { userData: IAPIUser; captchaToken: string },
+	): Promise<IAPISignupResponse> {
+		return this.authenticationService.signupUser(
+			{
+				...body.userData,
```

Comment:
whats this doing?

Reply:
that's the spread operator, basically copies all the data so that i can customize the ```_id``` parameter